jetty-websocket/javax-websocket-tests/src/main/java/org/eclipse/jetty/websocket/javax/tests/NetworkFuzzer.java

```diff
@@ -73,8 +77,22 @@ public NetworkFuzzer(LocalServer server, URI wsURI, Map<String, String> requestH
             HttpFields fields = this.upgradeRequest.getHeaders();
             requestHeaders.forEach((name, value) ->
             {
-                fields.remove(name);
-                fields.put(name, value);
+                if (HttpHeader.SEC_WEBSOCKET_SUBPROTOCOL.toString().equalsIgnoreCase(name))
+                {
+                    QuotedCSV subprotocols = new QuotedCSV(value);
+                    upgradeRequest.setSubProtocols(subprotocols.getValues().stream().collect(Collectors.toList()));
```

Comment:
What if there are multiple SEC_WEBSOCKET_SUBPROTOCOL headers?  wont each replace the last?   Why not define the QuotedCSV outside of the header loop?

Reply:
An offered list of WebSocket protocols is perfectly valid on the Request side.
Yes, this needs to be supported.